modules/pubCommonIdSystem.js

```diff
@@ -26,17 +45,41 @@ export const pubCommonIdSubmodule = {
   /**
    * performs action to obtain id
    * @function
-   * @returns {string}
+   * @param {SubmoduleParams} [configParams]
+   * @param {Object} [storedId] existing id
+   * @returns {IdResponse}
+   */
+  getId: function ({create = true, extend = false, pixelUrl} = {}, storedId) {
+    const response = {};
+    if (!storedId) {
+      const newId = (create) ? utils.generateUUID() : undefined;
+      response.id = newId;
+      response.callback = this.makeCallback(pixelUrl, newId);
+    } else if (extend) {
+      // When extending, only one of response fields is needed
+      const callback = this.makeCallback(pixelUrl, storedId);
+      if (callback) {
+        response.callback = callback;
+      } else {
+        response.id = storedId;
+      }
+    }
+
+    return response;
+  },
+  /**
+   * If the page already includes its own pubcid object, then delegate to that instead.
+   * @returns {(string|undefined)} Id if available.
    */
-  getId() {
+  getAltId: function() {
     // If the page includes its own pubcid object, then use that instead.
-    let pubcid;
     try {
       if (typeof window['PublisherCommonId'] === 'object') {
-        pubcid = window['PublisherCommonId'].getId();
+        // Only return the id if it's different from the stored id
+        return window['PublisherCommonId'].getId();
       }
-    } catch (e) {}
-    // check pubcid and return if valid was otherwise create a new id
-    return (pubcid) || utils.generateUUID();
+    } catch (e) {
+    }
+    return undefined;
```

Comment:
This line is unneeded as a function returns undefined by default if nothing was returned

Reply:
Good call.